popsycle/synthetic.py

```diff
@@ -79,8 +79,15 @@
            'ubv_i': 18, 'exbv': 19, 'obj_id': 20,
            'ubv_j': 21, 'ubv_u': 22, 'ubv_r': 23,
            'ubv_b': 24, 'ubv_h': 25, 'ubv_v': 26,
-           'teff': 27, 'grav': 28, 'mbol': 29, 'feh': 30 }
+           'teff': 27, 'grav': 28, 'mbol': 29, 'feh': 30,
+           'ztf_g': 31, 'ztf_r': 32}
```

Comment:
so there's always 31 and 32  no matter what, but if additional_photometric_systems is left as none, they are left empty?

Reply:
is this supposed to be hardcoded?

Reply:
ztf_g and ztf_r end up hardcoded into the `col_idx`, but they don't necessarily get used unless `additional_photometric_systems` isn't None. However it does mean that the hardcoded `col_idx` is incorrect in some real sense if you haven't added these columns. And it also means that there isn't really a way for this mechanism to work once we add more columns beyond ztf. It works here because `col_idx` is only used for specific columns at lower numbers in `calc_events`, and the rest of the entries in this dictionary are just for completeness.

Essentially having a hardcoded `col_idx` and allowing the user to add columns by argument are incompatible. I think this is why we should retire the col_idx system and instead move to an hdf5 data product that used compound datatypes instead. That way each hdf5 file would contain it's own information about how the columns are organized and we could future proof ourselves against an arbitrary number of photometric systems and filters.

Reply:
i thought we had talked about leaving everything as is for now, and only performing the conversion at the end, since it is not necessary to carry about ztf_g and ztf_r as they get calculated from the ubv quantities? or was the conclusion different?

Reply:
Jessica preferred that we put the ztf_g and ztf_r columns into the hdf5 file, saying that it was a very common operation for someone to want to compare the entire stellar population to their events and refined_events in the same filters.

Reply:
Of course these columns are only added to the hdf5 file (and events and refined_events files) if the user selects it by including 'ztf' in their additional_photometric_systems list. Otherwise these columns are left off the hdf5 file (and the events and refined_events thereafter).

Reply:
okay, then i must have just missed that 'if' statement for the hdf5 file

Reply:
This is indeed a temporary measure. `col_idx` will be gone in the next pull request.